sdk/azcore/runtime/poller.go

```diff
@@ -104,67 +103,82 @@ type NewPollerFromResumeTokenOptions[T any] struct {
 	// Response contains a preconstructed response type.
 	// The final payload will be unmarshaled into it and returned.
 	Response *T
+
+	// Operation[T] contains a custom polling implementation.
+	Operation Operation[T]
 }
 
 // NewPollerFromResumeToken creates a Poller from a resume token string.
 func NewPollerFromResumeToken[T any](token string, pl exported.Pipeline, options *NewPollerFromResumeTokenOptions[T]) (*Poller[T], error) {
 	if options == nil {
 		options = &NewPollerFromResumeTokenOptions[T]{}
 	}
-	tName, err := pollers.PollerTypeName[T]()
-	if err != nil {
+
+	if err := pollers.IsTokenValid[T](token); err != nil {
 		return nil, err
 	}
-	kind, err := pollers.KindFromToken(tName, token)
+	raw, err := pollers.ExtractToken(token)
 	if err != nil {
 		return nil, err
 	}
+	var asJSON map[string]interface{}
+	if err := json.Unmarshal(raw, &asJSON); err != nil {
+		return nil, err
+	}
+
+	opr := options.Operation
 	// now rehydrate the poller based on the encoded poller type
-	var lro pollers.Operation
-	switch kind {
-	case async.Kind:
-		log.Writef(log.EventLRO, "Resuming %s poller.", async.Kind)
-		lro = &async.Poller{}
-	case armloc.Kind:
-		log.Writef(log.EventLRO, "Resuming %s poller.", armloc.Kind)
-		lro = &armloc.Poller{}
-	case body.Kind:
-		log.Writef(log.EventLRO, "Resuming %s poller.", body.Kind)
-		lro = &body.Poller{}
-	case loc.Kind:
-		log.Writef(log.EventLRO, "Resuming %s poller.", loc.Kind)
-		lro = &loc.Poller{}
-	case op.Kind:
-		log.Writef(log.EventLRO, "Resuming %s poller.", op.Kind)
-		lro = &op.Poller{}
-	default:
-		return nil, fmt.Errorf("unhandled poller type %s", kind)
-	}
-	if err = json.Unmarshal([]byte(token), lro); err != nil {
+	if loc.CanResume(asJSON) {
+		log.Write(log.EventLRO, "Resuming loc poller.")
+		opr, _ = loc.New[T](pl, nil)
+	} else if op.CanResume(asJSON) {
+		log.Write(log.EventLRO, "Resuming op poller.")
+		opr, _ = op.New[T](pl, nil, "")
+	} else if opr != nil {
+		log.Writef(log.EventLRO, "Resuming custom poller %T.", opr)
+	} else {
+		return nil, fmt.Errorf("unhandled poller token %+v", raw)
+	}
+	if err := json.Unmarshal(raw, &opr); err != nil {
 		return nil, err
 	}
 	return &Poller[T]{
-		pt: pollers.NewPoller(lro, nil, pl),
-		rt: options.Response,
+		op:     opr,
+		result: options.Response,
 	}, nil
 }
 
+// Operation[T] abstracts the differences among long-running operation implementations.
+type Operation[T any] interface {
```

Comment:
This is the new piece of public surface area.  Open to other naming suggestions.

Reply:
Operation is used in C++ and .NET.  I like aligning on that name here as well.

Reply:
That was mostly why I chose the name.  The only thing here is that we use the term `Poller` instead of `Operation` for our LROs.  It would be nice if the term reflected that, but I didn't want to use something like `PollerImpl`.

Reply:
I'm fine with `Operation` but if you're looking for another name, `PollingMethod` seems reasonable too

Reply:
`PollingPolicy`
`PollerMethod`

Reply:
Eh, that suggests it's a `policy.Policy`

Reply:
Yeah I had the same thought.

How about `PollerMethod` then.  It's like `Poller` but with a suffix.